.openshift-ci/scripts/baseline/main.py

```diff
@@ -0,0 +1,335 @@
+                result.extend(values)
+                continue
+
+            # For every group (vm, method) sort the records in ascending order
+            # to get rid of the oldest data (at the beginning or the list).
+            ordered = sorted(values, key=get_timestamp)
+
+            # Verify for each test type whithin the data, that its length is
+            # not over the threshold. Drop the oldest items when needed.
+            for t in test_names:
+                tests = [item for item in ordered if is_test(t, item)]
+
+                if len(tests) >= threshold:
+                    cutoff = len(tests) - threshold
+                    result.extend(tests[cutoff:])
+                else:
+                    result.extend(tests)
+
+        result.extend(new_measurement)
+        return result
+
+
+def verify_data(data):
+    """
+    There are some assumptions made about the date we operate with for both
+    baseline series and new measurements, they're going to be verified below.
+    """
+
+    assert data, "Benchmark data must be not empty"
+
+    data_grouped = process(data)
+
+    for group, values in data_grouped:
+        counter = Counter()
+        for v in values:
+            counter.update(v.keys())
+
+        # Data provides equal number of with/without collector benchmark runs.
+        #
+        # NOTE: We rely only on baseline/collector hackbench at the moment. As
+        # soon as data for more tests will be collected, this section has to be
+        # extended accordingly.
+        no_overhead_count = counter.get("baseline_benchmark", 0)
+        overhead_count = counter.get("collector_benchmark", 0)
+
+        if (no_overhead_count != overhead_count):
+            raise Exception(f"Number of with/without overhead do not match:"
+                            f" {no_overhead_count}, {overhead_count} ")
+
+
+def intersection(baseline_data, new_data):
+    """
+    Remove any tests that are not found in both the baseline and the new data.
+
+    This is done for 2 reasons:
+    - Data that shows up in the baseline but not in the new data is considered a
+      removed test. There is no point in processing this.
+    - Data that shows up in the new data but not in the baseline is considered
+      a new test that doesn't have an associated baseline yet. The baseline for
+      the new test will be added after it is merged into master.
+    """
+    assert baseline_data, "Baseline data must be not empty"
+
+    baseline_grouped = process(baseline_data)
+    new_grouped = process(new_data)
+
+    new_keys = [items[0] for items in new_grouped]
+    baseline_filtered = [
+        data for data in baseline_grouped
+        if data[0] in new_keys
+    ]
+
+    baseline_keys = [items[0] for items in baseline_filtered]
+    new_filtered = [
+        data for data in new_grouped
+        if data[0] in baseline_keys
+    ]
+
+    return (baseline_filtered, new_filtered)
+
+
+def normalize_collection_method(method):
```

Comment:
I lied above. As it turns out I also needed to add this function to normalise the collection methods and some changes in `process` below to normalise VmConfig and collection method